### PR TITLE
Align directories and codegen with stock PyTorch

### PR DESCRIPTION
**Pull request by ZhiweiYan-96**
Draft version for now. Only for design review.

Can be built, can be tested with simple ut, like `test_xpu.py`.

**Not tested with other uts**
 
Added supports includes
- `REGISTER_XPU_DISPATCH(add_stub, add_kernel)`
- `TORCH_IMPL_FUNC`

### CODE REVIEW

**Review by dvrogozh — COMMENTED** [review]
If you are aligning project structure with pytorch, do you plan to move the code to the pytorch tree at some point? or this project will remain separate? If it will be separate, will it be reused elsewhere outside pytorch?

**Review by EikanWang — APPROVED** [review]

File: yaml/xpu_functions.yaml
Comment (EikanWang) [inline yaml/xpu_functions.yaml:1]: Is this file still valid ?

File: yaml/native/native_functions.yaml [hunk 1]
```diff
@@ -3335,4 +3335,308 @@
...
+  variants: function
+  structured_delegate: linalg_cross.out
+  dispatch:
+    ZeroTensor: linalg_cross_zerotensor
+
+- func: linalg_cross.out(Tensor self, Tensor other, *, int dim=-1, Tensor(a!) out) -> Tensor(a!)
+  python_module: linalg
+  structured: True
+  dispatch:
+    CPU, CUDA, MPS: linalg_cross_out
```
Comment (EikanWang) [inline yaml/native/native_functions.yaml]: @ZhiweiYan-96 , could you help elaborate on this line of code? It is irrelevant to Intel GPU. Right?

File: yaml/xpu_functions.yaml
Comment (ZhiweiYan-96) [reply yaml/xpu_functions.yaml:1]: nope, need removed

File: yaml/native/native_functions.yaml (hunk 1 above)
Comment (ZhiweiYan-96) [reply yaml/native/native_functions.yaml]: sure, I miss this line, I would removed it

**Comment by fengyuan14** [comment]
Please emend the description of the PR, to give formal description on the feature.

**Pull request merged**